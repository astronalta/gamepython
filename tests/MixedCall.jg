# jgtest exec on

mixed4(a Int, b Float, c Int, d Float) {
    Io::println(a.str)
    Io::println(b.str)
    Io::println(c.str)
    Io::println(d.str)
}

mixed20(a Int, b Float, c Int, d Float, e Int, f Float, g Int, h Float, i Int, j Float,
        k Int, l Float, m Int, n Float, o Int, p Float, q Int, r Float, s Int, t Float) {

    Io::println(a.str)
    Io::println(b.str)
    Io::println(c.str)
    Io::println(d.str)
    Io::println(e.str)
    Io::println(f.str)
    Io::println(g.str)
    Io::println(h.str)
    Io::println(i.str)
    Io::println(j.str)
    Io::println(k.str)
    Io::println(l.str)
    Io::println(m.str)
    Io::println(n.str)
    Io::println(o.str)
    Io::println(p.str)
    Io::println(q.str)
    Io::println(r.str)
    Io::println(s.str)
    Io::println(t.str)
}

main() Int {
    mixed4(1, 0.2, 3, 0.4)
    Io::println("")
    mixed20(1, 0.2, 3, 0.4, 5, 0.6, 7, 0.8, 9, 1.0, 
            11, 1.2, 13, 1.4, 15, 1.6, 17, 1.8, 19, 2.0)

    ret 0
}
