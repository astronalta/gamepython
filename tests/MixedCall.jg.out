1
0.2
3
0.4

1
0.2
3
0.4
5
0.6
7
0.8
9
1
11
1.2
13
1.4
15
1.6
17
1.8
19
2
